lib/child_process.js

```diff
@@ -247,37 +255,42 @@ exports.execFile = function(file /*, args, options, callback*/) {
     }, options.timeout);
   }
 
-  child.stdout.addListener('data', function(chunk) {
-    stdoutLen += chunk.length;
-
-    if (stdoutLen > options.maxBuffer) {
-      ex = new Error('stdout maxBuffer exceeded');
-      kill();
-    } else {
-      if (!encoding)
-        _stdout.push(chunk);
-      else
-        _stdout += chunk;
-    }
-  });
-
-  child.stderr.addListener('data', function(chunk) {
-    stderrLen += chunk.length;
-
-    if (stderrLen > options.maxBuffer) {
-      ex = new Error('stderr maxBuffer exceeded');
-      kill();
-    } else {
-      if (!encoding)
-        _stderr.push(chunk);
-      else
-        _stderr += chunk;
-    }
-  });
+  if (child.stdout) {
+    if (encoding)
+      child.stdout.setEncoding(encoding);
```

Comment:
No. I just moved the code up a few lines to avoid another `if (child.stdout)` check.